core/unit_test/cuda/TestCuda_InterOp_GraphMultiGPU.cpp

```diff
@@ -0,0 +1,116 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include "Kokkos_Core.hpp"
+#include "Kokkos_Graph.hpp"
+#include "TestCuda_Category.hpp"
+
+namespace {
+
+class TEST_CATEGORY_FIXTURE(multi_gpu) : public ::testing::Test {
+ public:
+  using graph_t          = Kokkos::Experimental::Graph<Kokkos::Cuda>;
+  using graph_node_ref_t = Kokkos::Experimental::GraphNodeRef<Kokkos::Cuda>;
+
+ public:
+  void SetUp() override {
+    KOKKOS_IMPL_CUDA_SAFE_CALL(cudaGetDeviceCount(&num_devices));
+
+    if (num_devices < 2)
+      GTEST_SKIP()
```

Comment:
Is it actually legal to skip in the setup member function of the fixture?

Reply:
Seems it is legal from https://google.github.io/googletest/reference/testing.html#GTEST_SKIP:
> Can be used in individual test cases or in the SetUp() methods of test environments or test fixtures